pkg/perfcounter/context.go

```diff
@@ -32,11 +32,27 @@ func WithCounterSet(ctx context.Context, sets ...*CounterSet) context.Context {
 		}
 		return context.WithValue(ctx, CtxKeyCounters, v)
 	}
+
 	counters := v.(CounterSets)
+
+	allExist := true
```

Comment:
what is the expected cost of this loop?    How often we call this, and any expected number of loops?